machinery/topology.go

```diff
@@ -61,12 +63,19 @@ func WithLinks(links ...LinkFunc) TopologyOptionsFunc {
 	}
 }
 
+// WithoutEnsureDAG allows the creation of a topology that may contain loops
+func WithoutEnsureDAG() TopologyOptionsFunc {
+	return func(o *TopologyOptions) {
+		o.EnsureDag = false
```

Comment:
```suggestion
		o.AllowLoops = true
```

Then no need to set the default in `func NewTopology(...)` anymore.